### ISSUE TITLE
update: Data path ".from" should be string.

### ISSUE THREAD

**Issue by Splaktar** [angular/angular-cli#10312]
### Versions
```
Angular CLI: 6.0.0-rc.4
Node: 8.10.0
OS: darwin x64
Angular: 6.0.0-rc.4
... animations, cli, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, platform-server, router
... service-worker

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.5.5
@angular-devkit/build-angular     0.5.5
@angular-devkit/build-optimizer   0.5.5
@angular-devkit/core              0.5.5
@angular-devkit/schematics        0.5.6
@angular/cdk                      6.0.0-rc.2
@angular/flex-layout              5.0.0-beta.13
@angular/material                 6.0.0-rc.2
@ngtools/json-schema              1.1.0
@ngtools/webpack                  6.0.0-rc.3
@schematics/angular               0.5.6
@schematics/update                0.5.6
rxjs                              5.5.10
typescript                        2.7.2
webpack                           4.5.0
```

### Repro steps
* Step 1 - Install CLI rc.4 and see this message
```
================================================================================
An old CLI configuration has been detected, which needs to be updated to the latest version.

Please run the following command to update this workspace:
  ng update @angular/cli --migrate-only --from=1
================================================================================
```
* Step 2 - Run `ng update @angular/cli --migrate-only --from=1` as suggested and get this error
```
Schematic input does not validate against the Schema: {"dryRun":false,"migrateOnly":true,"from":1,"force":false,"all":false,"next":false,"registry":"http://registry.npmjs.org/"}
Errors:

  Data path ".from" should be string.
```
* Step 3 - Run `ng update @angular/cli --migrate-only --from="1"` and get the same error
* Step 4 - Run `ng update @angular/cli --migrate-only --from='1'` and get the same error

### Desired behavior
The command recommended by the CLI message should run without error.

### Mention any other details that might be useful (optional)
N/A

**Comment by dkaushikl** [comment]
`ng update @angular/cli --migrate-only --from=1.7.4` not work

**Comment by benouat** [comment]
The nodejs module used behind the scene to get command line parameters is probably wrongly casting `--from=1` to boolean instead of keeping it as a string.

`ng update @angular/cli --migrate-only --from=1.7.4` just worked for me

**Comment by nik-base** [comment]
Use your previous version of angular-cli instead of 1 for the --from attribute.
So it would look something like this.
`ng update @angular/cli --migrate-only --from=<previous_version_of_cli>`
The new angular cli doesnt have `angular-cli.json`, it is replaced by `angular.json`.

**Comment by filipesilva** [comment]
Yes it looks like our arg parsing isn't reading `--from=1` as a number. There is a fix incoming in https://github.com/angular/angular-cli/pull/10315. Sorry about that :/

**Comment by dkaushikl** [comment]
after above command, it generates a new error while build or serve application
Cannot read property 'config' of null

**Comment by mantonik** [comment]
Hi, as I understand fix is in new build, but what I should do now to move forward?

When I do 
ng update @angular/cli --migrate-only 
    Updating package.json with dependency @angular/cli @ "1.7.4" (was "1.5.0")...

but ng server still is giving a problem-related cli version 

I tried 
[mariusz admin.angular5]$ ng update @angular/cli --migrate-only --from=1.5.0
/Users/mariusz/dev/selenium_testing_portal/projects/admin.angular5>
[mariusz admin.angular5]$ ng update @angular/cli --migrate-only --from="1.5.0"
/Users/mariusz/dev/selenium_testing_portal/projects/admin.angular5>

but those look like no effect 

my current config 

[mariusz admin.angular5]$ ng --version 

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 6.0.0-rc.4
Node: 8.11.1
OS: darwin x64
Angular: 5.2.9
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.5.6 (cli-only)
@angular-devkit/core         0.5.6 (cli-only)
@angular-devkit/schematics   0.5.6 (cli-only)
@angular/cli                 <error>
@schematics/angular          0.5.6 (cli-only)
@schematics/update           0.5.6 (cli-only)
rxjs                         5.5.10
typescript                   2.7.2 (cli-only)
    
Thank you for suggestion

**Comment by Splaktar** [comment]
@mantonik I would try `ng update @angular/cli --migrate-only --from=1.7.4` and if that doesn't work, try upgrading to 1.7.4, getting things working, and then re-running the update command with 6.0.0-rc.4.

**Comment by Brocco** [comment]
The fix referenced (#10315) has not yet been released and because both `1` and `1.7` can both be parsed as numbers that is what is happening. The fix will respect the option's declared type and parse those values as strings even though they *could* be parsed as numbers.

Thanks for your vigilance in testing these issues.
